src/components/Combobox/Combobox.jsx

```diff
@@ -0,0 +1,215 @@
+
+		let stateValue;
+
+		if (typeof value !== 'undefined') {
+			stateValue = value;
+		} else if (typeof defaultValue !== 'undefined') {
+			stateValue = defaultValue;
+		}
+
+		if (stateValue) {
+			const valueChild = React.Children.toArray(children).find(child => child.props.value === stateValue);
+
+			this.state = {
+				value: stateValue
+			};
+
+			if (valueChild) {
+				this.state = {
+					...this.state,
+					value: valueChild.props.children,
+					selectboxValue: valueChild.props.value
+				};
+			}
+		}
+	}
+
+	componentWillReceiveProps(newProps) {
+		const {defaultValue} = newProps;
+		const children = React.Children.toArray(newProps.children);
+
+		const newPropsContainNewValue = typeof newProps.value !== 'undefined';
+
+		if (newPropsContainNewValue) {
+			const child = children.find(child => child.props.value === newProps.value);
+			this.setState({
+				value: (child && child.props.children) || newProps.value,
+				selectboxValue: child && child.props.value
+			});
+		} else {
+			const previousValueIsReset = typeof this.props.value !== 'undefined' && !newPropsContainNewValue;
+			const currentValue = this.state.value;
+
+			if (previousValueIsReset) {
+				this.setState({
+					value: (void 0), //eslint-disable-line no-void
+					selectboxValue: (void 0) //eslint-disable-line no-void
+				});
+			}
+
+			const hasCurrentValue = typeof currentValue !== 'undefined' && currentValue !== '';
+			const newPropsContainNewDefaultValue = typeof newProps.defaultValue !== 'undefined';
+
+			const shouldSetNewDefaultValue =
+				(previousValueIsReset || !hasCurrentValue) &&
+				newPropsContainNewDefaultValue;
+
+			if (shouldSetNewDefaultValue) {
+				const defaultChild = children.find(child => child.props.value === defaultValue);
+				this.setState({
+					value: (defaultChild && defaultChild.props.children) || defaultValue,
+					selectboxValue: defaultChild && defaultChild.props.value
+				});
+			}
+		}
+	}
+
+	render() {
+		const {
+			theme,
+			placeholder,
+			isDisabled,
+			caretIconName,
+			selectedItemIconName,
+			InputComponent: Input,
+			children,
+			SelectboxComponent: Selectbox
+		} = this.props;
+
+		const {value, selectboxValue} = this.state;
+
+		const inputProps = {
+			type: 'text',
+			value: value || '',
+			onChange: this.onChange,
+			onBlur: this.onBlur,
+			disabled: isDisabled,
+			theme: {
+				container: theme.input
+			}
+		};
+
+		if (!value && placeholder) {
+			inputProps.placeholder = placeholder;
+		}
+
+		const selectboxProps = {
+			value: selectboxValue,
+			placeholder: placeholder || '',
+			caretIconName,
+			selectedItemIconName,
+			children,
+			isDisabled,
+			theme,
+			onChange: this.onChangeSelectbox
+		};
+
+		const className = classnames(theme.container, {
+			[theme.container_isDisabled]: isDisabled
+		});
+
+		return (
+			<div className={className}>
+				<div className={theme.selectbox}>
+					<Selectbox {...selectboxProps} />
+				</div>
+				<Input {...inputProps}/>
```

Comment:
Popover calculate position relative anchor. If anchor be less than container in which it located input,  popover will have wrong position.